aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -286,19 +372,19 @@ module AptosFramework::Coin {
         deposit(to, coin);
     }
 
-    /// Returns the `value` of the passed in `coin`.
+    /// Returns the `value` passed in `coin`.
     public fun value<CoinType>(coin: &Coin<CoinType>): u64 {
         coin.value
     }
 
-    /// Withdraw specifed `amount` of coin from the signing account.
+    /// Withdraw specifed `amount` of coin `CoinType` from the signing account.
     public fun withdraw<CoinType>(
         account: &signer,
         amount: u64,
     ): Coin<CoinType> acquires CoinStore {
         let account_addr = Signer::address_of(account);
         assert!(
-            exists<CoinStore<CoinType>>(account_addr),
+            is_registered_for_account<CoinType>(account_addr),
```

Comment:
we need to emit an event for every transaction that pays a gas fee here, I was trying to avoid it

Reply:
Yes this is one aspect I'm thinking through at the moment. Would love to hear your thoughts on it. To summarize the discussion so far:

Gas burns also generate withdrawal events since TransactionFees calls Coin::withdraw, which in my opinion is acceptable and actually could be useful. If we strongly don't want that from an UX perspective, we'd need to either 
(1) add a functionality to burn coins from an account (with burn capability) or
(2) have an option to allow withdrawal without generating events.

I'm leaning more toward (1) if we want to do that since (2) can open up the door for abuse (“secret” withdrawals that don't generate events).

Reply:
I updated with approach (1), adding a burn_from_account function to Coin.move that allows someone with BurnCapability to burn coins directly from an account. We can discuss later on whether it makes sense for this functionality to be in the standard Coin vs being in a different extended Coin version (e.g. BurnableCoin)